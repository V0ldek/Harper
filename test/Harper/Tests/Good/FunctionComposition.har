fac :: Integer -> Integer;
fac n = {
  var (result :: Integer) = 1;
  var (i :: Integer) = 0;
  while i < n {
    i += 1;
    result *= i;
  }
  return result;
};

sqr :: Integer -> Integer;
sqr n = n * n;

facSqr :: Integer -> Integer;
facSqr = sqr @ fac;

main :: Integer;
main = facSqr 10;