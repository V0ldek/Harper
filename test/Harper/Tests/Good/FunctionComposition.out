Execution ended with value: 13168189440000